many probabilistic doppyo metrics (#439)
pangeo-data/climpred: CHANGELOG.rst

  (:issue:`352`, :pr:`418`) `Aaron Spring`_.
- ``PredictionEnsemble.verify()`` does not correct ``dim`` automatically to
  ```member`` for probabilistic metrics. (:pr:`407`) `Aaron Spring`_.

New Features
------------

- Use math operations like ``+-*/`` with :py:class:`~climpred.classes.HindcastEnsemble`
  and :py:class:`~climpred.classes.PerfectModelEnsemble`. (:pr:`377`) `Aaron Spring`_.
- plot all datasets in :py:class:`~climpred.classes.HindcastEnsemble` or
  :py:class:`~climpred.classes.PerfectModelEnsemble` by

>   :py:func:`~climpred.classes.PredictionEnsemble.plot` if no other spatial dimensions

Author: ```suggestion
  :py:meth:`~climpred.classes.PredictionEnsemble.plot` if no other spatial dimensions
```

  are present. (:pr:`383`) `Aaron Spring`_.
- Assertion functions for :py:class:`~climpred.classes.PerfectModelEnsemble`:
  ``climpred.testing.assert_PredictionEnsemble``. (:pr:`391`) `Aaron Spring`_.

> - :py:class:`~climpred.classes.HindcastEnsemble.bootstrap` analogous to

Author: ```suggestion
- :py:meth:`~climpred.classes.HindcastEnsemble.bootstrap` analogous to
```

>   :py:class:`~climpred.classes.PerfectModelEnsemble.bootstrap`.

Author: ```suggestion
  :py:meth:`~climpred.classes.PerfectModelEnsemble.bootstrap`.
```

  (:issue:`257`, :pr:`418`) `Aaron Spring`_.

> - ``PredictionEnsemble.verify()`` allows all dimensions from `initialized` as ``dim``.

Author: ```suggestion
- :py:meth:`~climpred.HindcastEnsemble.verify` allows all dimensions from `initialized` as ``dim``.
```

  For spatial dimensions to be used with ``skipna=True`` when masked input data.
  (:issue:282, :pr:`407`) `Aaron Spring`_.

> - Allow binary verif and (un)init forecasts as probabilities in ``hindcast.verify()``

Author: ```suggestion
- Allow binary verif and (un)init forecasts as probabilities in :py:meth:`~climpred.HindcastEnsemble.verify`
```

  without providing ``logical``, so

>   ``hindcast.map(logical).mean('member').verify(metric='brier_score', comparison='e2o',dim=[])==``

Author: ```suggestion
  ``hindcast.map(logical).verify(metric='brier_score', comparison='m2o',dim=“member”==``
```
